src/coreComponents/constitutive/capillaryPressure/JFunctionCapillaryPressure.cpp

```diff
@@ -250,7 +250,8 @@ void JFunctionCapillaryPressure::saveConvergedRockState( arrayView2d< real64 con
     {
       permeability = convergedPermeability[ei][0][2];
     }
-    GEOS_ERROR_IF( permeability < LvArray::NumericLimits< real64 >::epsilon, "Zero permeability in J-function capillary pressure" );
+    // 9.869233×10−13 is Darcy to sq m factor
+    GEOS_ERROR_IF( permeability < LvArray::NumericLimits< real64 >::epsilon * 9.869233e-13, "Zero permeability in J-function capillary pressure" );
```

Comment:
why is this error check even necessary? A zero permeability is a valid number and all underflows are flushed to zero.

Reply:
there is a division down here https://github.com/GEOS-DEV/GEOS/blob/6ebf57643a6fc6136f9687cfbc902bc662fc39a2/src/coreComponents/constitutive/capillaryPressure/JFunctionCapillaryPressure.cpp#L267